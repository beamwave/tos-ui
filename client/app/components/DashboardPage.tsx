import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import ReactGridGallery from 'react-grid-gallery'
import { FontAwesomeIcon } from '../../../node_modules/@fortawesome/react-fontawesome'
import { check, deepEquals } from '../helpers/helpers'
import { startSearch } from '../actions/school'
import { throttle, debounce } from 'throttle-debounce'

// gradeAdjustmentStartDt = invoice start date

// invoice button display requirements: tests are shipped (not pending), correct invoice window, otherwise show invoice

// allow edit if invoice window still opened

// react acl

// public search is possible from any react page

// student names, prevent invoice generation

// fee waiver from order 30 (fee waivers available)

// data for students taking current test

// yearly changes to names, ie []

// order table: aicode (partition?), testDt, eventid (sort?), weborderid (index?), { lines } ?,

// get orders by aicode and possibly filter by year

// learn about aws playground

// get sample data for hyperloop events (orders, invoice, user)

// current possible tables: misgrid adjustments, orders, test takers, ship confirmation, reference (test date, window dates, event ids)

// add lambda tag: key: 'PERSISTENT': 'YES'

interface IProps {
  startSearch: (data: any) => any[]
  results: any
  // username: string
  // category: string
  // categories?: any
  // images: any[]
  // filters?: any[]
  // filter?: (by: FData) => any
  // history?: any
}

interface IState {
  // images: any[]
}

export class DashboardPage extends Component<IProps> {
  state = {
    state: 'all',
    filterType: 'school',
    stateCollege: 'bard college',
    districtCollege: 'bard college',
    stateContractCoverage: 'all',
    districtContractCoverage: 'state',
    schoolQuery: { active: true, query: '' },
    districtQuery: { active: false, query: '' },
    results: []
  }

  // componentWillReceiveProps = (nextProps, nextState) => {
  //   const { results } = nextProps
  //   this.setState({ results })
  // }

  onFieldChange = ({ target: { name, value } }) =>
    this.setState({ [name]: value } as any)

  onInputChange = ({ target: { name, value } }) =>
    this.setState({ [name]: { active: true, query: value } } as any)

  onFilterChange = ({ target: { name, value } }) => {
    // set filterType value
    this.setState({ [name]: value } as any)

    // set active state of queries
    if (value === 'school')
      this.state.districtQuery = { active: false, query: '' }

    if (value === 'district')
      this.state.schoolQuery = { active: false, query: '' }
  }

  onQuery = async userInput => this.props.startSearch(userInput)

  // call action creator 1 second after user finishes typing
  startSearchDebounce = debounce(1000, this.onQuery)

  // call action creator every 750 milliseconds during typing
  startSearchThrottle = throttle(750, this.onQuery)

  // store user input in local state ??
  onUserInput = userInput => this.setState({ userInput })

  // query db when user begins searching
  onSubmit = async e => {
    e.preventDefault()
    // const {
    //   target: { value: userInput }
    // } = e
    const { filterType } = this.state

    const data = {
      userInput: '',
      filters: {}
    }

    if (filterType === 'school') {
      const {
        schoolQuery: { query }
      } = this.state
      data.userInput = query
    }
    if (filterType === 'district') {
      const {
        districtQuery: { query },
        state,
        districtCollege,
        districtContractCoverage
      } = this.state
      data.userInput = query
      data.filters = { state, districtCollege, districtContractCoverage }
    }
    if (filterType === 'state') {
      const { stateCollege, stateContractCoverage } = this.state
      data.filters = { stateCollege, stateContractCoverage }
    }

    const { userInput } = data

    // start querying db at various intervals
    if (userInput.length < 5) this.startSearchThrottle(data)
    else this.startSearchDebounce(data)
  }

  displaySchoolInput = () => (
    <form onSubmit={this.onSubmit} className="school-input-container">
      <select
        name="filterType"
        className="searchbar-select"
        onChange={this.onFilterChange}
        value={this.state.filterType}
      >
        <option value="school">school</option>
        <option value="district">district</option>
        <option value="state">state</option>
      </select>
      <input
        type="text"
        name="schoolQuery"
        className="searchbar-input"
        value={this.state.schoolQuery.query}
        onChange={this.onInputChange}
        placeholder="search school by name or code"
      />
      <button className="blue-button">Search</button>

      <p className="link">Advanced Search</p>
    </form>
  )

  displayDistrictInput = () => (
    <form onSubmit={this.onSubmit} className="district-input-container">
      <select
        name="filterType"
        className="searchbar-select"
        value={this.state.filterType}
        onChange={this.onFilterChange}
      >
        <option value="school">school</option>
        <option value="district">district</option>
        <option value="state">state</option>
      </select>
      <input
        type="text"
        name="districtQuery"
        className="searchbar-input"
        value={this.state.districtQuery.query}
        onChange={this.onInputChange}
        placeholder="District Name, Code"
      />

      <p>in</p>
      <select
        name="state"
        className="searchbar-select"
        value={this.state.state}
        onChange={this.onFieldChange}
      >
        <option value="all">All</option>
        <option value="aa">AA</option>
        <option value="ae">AE</option>
        <option value="alabama">Alabama</option>
      </select>

      <select
        name="districtCollege"
        className="searchbar-select"
        value={this.state.districtCollege}
        onChange={this.onFieldChange}
      >
        <option value="bard college">Bard College</option>
        <option value="furman university">Furman University</option>
        <option value="harvard college">Harvard College</option>
        <option value="mercer university">Mercer University</option>
      </select>

      <p>Contract Coverage</p>
      <select
        name="stateContractCoverage"
        className="searchbar-select"
        value={this.state.stateContractCoverage}
        onChange={this.onFieldChange}
      >
        <option value="all">All</option>
        <option value="district">District</option>
        <option value="state">State</option>
        <option value="none">None</option>
      </select>

      <button className="blue-button">Search</button>
    </form>
  )

  displayStateInput = () => (
    <form onSubmit={this.onSubmit} className="state-input-container">
      <select
        name="filterType"
        className="searchbar-select"
        onChange={this.onFieldChange}
        value={this.state.filterType}
      >
        <option value="school">school</option>
        <option value="district">district</option>
        <option value="state">state</option>
      </select>

      <select
        name="stateCollege"
        className="searchbar-select"
        onChange={this.onFieldChange}
        value={this.state.stateCollege}
      >
        <option value="bards college">Bards College</option>
        <option value="charles r drew">
          Charles R. Drew University of Medicine and Science
        </option>
        <option value="dod">Department of Defense Education Activity</option>
      </select>

      <p>Contract Coverage</p>
      <select
        name="districtContractCoverage"
        className="searchbar-select"
        value={this.state.districtContractCoverage}
        onChange={this.onFieldChange}
      >
        <option value="state">State</option>
      </select>

      <button className="blue-button">View</button>
    </form>
  )

  render = () => {
    const { filterType } = this.state
    const { results } = this.props

    return (
      <div className="dashboard-container">
        <nav>
          <h2>Test Ordering</h2>
          <div>
            <p>Org Search</p>
            <p>Calendar</p>
            <p>Help</p>
          </div>
        </nav>
        <header>
          {filterType === 'school' && this.displaySchoolInput()}
          {filterType === 'district' && this.displayDistrictInput()}
          {filterType === 'state' && this.displayStateInput()}
        </header>
        <main>
          {results.length > 0 && (
            <h3>
              {results.length} Search Results for "{results[0].query}"
            </h3>
          )}
          <div className="results-table">
            {results &&
              results.map(
                ({
                  aicode,
                  schoolName,
                  schoolAddress,
                  schoolStatus,
                  schoolAdmins
                }) => [
                  <header key={`${aicode}-${schoolAddress}`}>
                    <p>School Name</p>
                    <p>Address</p>
                    <p>Status</p>
                    <p>Adminstration</p>
                    <div>
                      <p>Contract</p>
                      <div className="subhead">
                        <p>State</p>
                        <p>Distr.</p>
                      </div>
                    </div>
                    <p>Study</p>
                    <p>Research Group</p>
                  </header>,
                  <main key={`${aicode}-${schoolName}`}>
                    <div>
                      <p>{schoolName}</p>
                      <p>AI {aicode}</p>
                    </div>
                    <div>
                      <p>{schoolAddress}</p>
                    </div>
                    <div>
                      <p>{schoolStatus}</p>
                    </div>
                    <div className="admins">
                      {schoolAdmins.map(
                        ({ admin, date, contract, study, research }) => (
                          <div className="admin-row" key={`${admin}-${date}`}>
                            <div>
                              <p>{admin}</p>
                              <p className="date">{date}</p>
                            </div>
                            <p>{contract}</p>
                            <p>{study}</p>
                            <p>{research}</p>
                          </div>
                        )
                      )}
                    </div>
                    <div className="view-link">
                      <Link to="/school-landing">View</Link>
                    </div>
                  </main>

                ]
              )}
          </div>
          {/* <table>
            {results &&
              results.map(
                ({
                  aicode,
                  schoolName,
                  schoolAddress,
                  schoolStatus,
                  schoolAdmins
                }) => [
                  <thead>
                    <tr>
                      <th>School Name</th>
                      <th>Address</th>
                      <th>Status</th>
                      <th>Adminstration</th>
                      <th>Contract</th>
                      <th>Study</th>
                      <th>Research Group</th>
                    </tr>
                  <thead>,
                  <tbody>
                    <tr>
                      <td>
                        <tr>{schoolName}</tr>
                        <tr>AI {aicode}</tr>
                      </td>
                      <td>{schoolAddress}</td>
                      <td>{schoolStatus}</td>
                      <td colSpan={3}>
                        {schoolAdmins.map(
                          ({ admin, date, contract, study, research }) => [
                            <tr>
                              <td>
                                {admin}
                                {'\n'}
                                {date}
                              </td>
                              <td>{contract}</td>
                              <td>{study}</td>
                              <td>{research}</td>
                            </tr>
                          ]
                        )}
                      </td>
                    </tr>
                  </tbody>
                ]
              )}
          </table> */}
        </main>
      </div>
    )
  }
}

// aicode: action.results.aicode,
// schoolName: action.results.schoolName,
// schoolAddress: action.results.schoolAddress,
// schoolStatus: action.results.schoolStatus,
// schoolAdmins: action.results.schoolAdmins

const mapStateToProps = (state, props) => ({
  results: state.school.results
  // token: state.auth.token,
  // username: state.auth.username,
  // category: state.app.category,
  // categories: state.images.categories,
  // images: state.images.all,
  // filters: state.filters
})

export default connect<any>(
  mapStateToProps,
  { startSearch }, // enables use of startSearch action creator
  null,
  { pure: false }
)(DashboardPage)

// state = {
//   images: [],
//   categories: 'map an object'
// }

// filter images
// handleFilterChange = ({ target }) => {
//   const { filters, filter } = this.props
//   // "medium", "device", "wardrobe", "mood", etc.
//   // console.log('dataset', target.dataset.filterName)
//   // console.log('value', target.value)

//   const by = {
//     category: target.dataset.filterName,
//     value: target.value
//   }

//   filter(by)
// }

// handleThumbnailClick = (i, e) => {
//   const { images, history } = this.props
//   const imageId = images[i]._id
//   history.push(`/image/${imageId}`)
// }

// componentDidMount = () => {
//   const { images: imagesFromServer } = this.props
//   // console.log('imagesFromServer', imagesFromServer)
//   let images
//   if (check(imagesFromServer) && Array.isArray(imagesFromServer))
//     images = imagesFromServer.map(image => ({
//       src: image.url,
//       thumbnail: image.url,
//       thumbnailHeight: image.height,
//       thumbnailWidth: image.width
//     }))

//   this.setState({ images })
// }

// componentWillReceiveProps = (nextProps, nextState) => {
//   if (
//     check(nextProps.images) &&
//     Array.isArray(nextProps.images) &&
//     (check(this.state.images) && Array.isArray(this.state.images))
//   ) {
//     if (!deepEquals(nextProps.images, this.state.images.length)) {
//       // if (nextProps.images.length !== this.state.images.length) {
//       let { images } = nextProps
//       images = images.map(image => ({
//         src: image.url,
//         thumbnail: image.url,
//         thumbnailHeight: image.height,
//         thumbnailWidth: image.width
//       }))

//       this.setState({ images })
//     }
//   }
// }

// render = () => {
//   const {
//     category,
//     categories,
//     images: imagesFromServer,
//     filters
//   } = this.props
//   const { images } = this.state
//   const alive = check(imagesFromServer)
//   return (
//     <main className="dashboard-page">
//       {
//         <div>
//           <section className="image-results">
//             {alive &&
//               imagesFromServer.length === 0 && (
//                 <p className="null-case">No images to display.</p>
//               )}
//             {alive &&
//               imagesFromServer.length > 0 && (
//                 <ReactGridGallery
//                   images={check(imagesFromServer) ? images : []}
//                   enableImageSelection={false}
//                   onClickThumbnail={this.handleThumbnailClick}
//                 />
//               )}
//           </section>
//           <aside className="primary-config">
//             {category === 'art' && (
//               <main>
//                 <form className="filter-container">
//                   <label htmlFor="select">medium</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['medium']}
//                     data-filter-name="medium"
//                     onChange={this.handleFilterChange}
//                   >
//                     {categories !== undefined &&
//                     categories.hasOwnProperty('medium') ? (
//                       categories['medium'].map(option => (
//                         <option key={option.name} value={option.name}>
//                           {option.name}
//                         </option>
//                       ))
//                     ) : (
//                       <option value="n/a">n/a</option>
//                     )}
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">device</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['device']}
//                     data-filter-name="device"
//                     onChange={this.handleFilterChange}
//                   >
//                     {categories !== undefined &&
//                     categories.hasOwnProperty('device') ? (
//                       categories['medium'].map(option => (
//                         <option key={option.name} value={option.name}>
//                           {option.name}
//                         </option>
//                       ))
//                     ) : (
//                       <option value="n/a">n/a</option>
//                     )}
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">project</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['project']}
//                     data-filter-name="project"
//                     onChange={this.handleFilterChange}
//                   >
//                     {categories !== undefined &&
//                     categories.hasOwnProperty('device') ? (
//                       categories['medium'].map(option => (
//                         <option key={option.name} value={option.name}>
//                           {option.name}
//                         </option>
//                       ))
//                     ) : (
//                       <option value="n/a">n/a</option>
//                     )}
//                   </select>
//                 </form>
//               </main>
//             )}
//             {category === 'comments' && (
//               <main>
//                 <form className="filter-container">
//                   <label htmlFor="select">topic</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['topic']}
//                     data-filter-name="topic"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">style</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['style']}
//                     data-filter-name="style"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">site</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['site']}
//                     data-filter-name="site"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//               </main>
//             )}
//             {category === 'clothes' && (
//               <main>
//                 <form className="filter-container">
//                   <label htmlFor="select">wardrobe</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['wardrobe']}
//                     data-filter-name="wardrobe"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">attire</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['attire']}
//                     data-filter-name="attire"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//               </main>
//             )}
//             {category === 'people' && (
//               <main>
//                 <form className="filter-container">
//                   <label htmlFor="select">mood</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['mood']}
//                     data-filter-name="mood"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">group</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['group']}
//                     data-filter-name="group"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">angle</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['angle']}
//                     data-filter-name="angle"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//               </main>
//             )}
//           </aside>
//           <aside className="secondary-config">
//             {category === 'art' && (
//               <main>
//                 <div className="filter-group">
//                   <div className="input-group">
//                     <label htmlFor="color">color</label>
//                     <input
//                       className="input"
//                       name="color"
//                       type="text"
//                       placeholder="#hex or name"
//                     />
//                     <div className="color-group">
//                       <div className="color-1" />
//                       <div className="color-2" />
//                       <div className="color-3" />
//                       <FontAwesomeIcon icon="angle-right" className="icon" />
//                     </div>
//                   </div>
//                 </div>
//                 <div className="filter-group">
//                   <div className="input-group">
//                     <label htmlFor="length">length</label>
//                     <input name="length" type="text" placeholder="px" />
//                   </div>
//                   <div className="input-group">
//                     <label htmlFor="width">width</label>
//                     <input name="width" type="text" placeholder="px" />
//                   </div>
//                 </div>
//               </main>
//             )}
//             {category === 'comments' && (
//               <main>
//                 <div className="filter-group">
//                   <div className="input-group">
//                     <label htmlFor="username">username</label>
//                     <input className="names" name="username" type="text" />
//                   </div>
//                 </div>
//               </main>
//             )}
//             {category === 'clothes' && (
//               <main>
//                 <div className="filter-group">
//                   <div className="input-group">
//                     <label htmlFor="color">color</label>
//                     <input
//                       className="input"
//                       name="color"
//                       type="text"
//                       placeholder="#hex or name"
//                     />
//                     <div className="color-group">
//                       <div className="color-1" />
//                       <div className="color-2" />
//                       <div className="color-3" />
//                       <FontAwesomeIcon icon="angle-right" className="icon" />
//                     </div>
//                   </div>
//                 </div>
//                 <div className="filter-group gender">
//                   <div className="input-group">
//                     <label>gender</label>
//                     <FontAwesomeIcon icon="male" className="male-icon" />
//                   </div>
//                   <div className="input-group">
//                     <FontAwesomeIcon icon="female" className="female-icon" />
//                   </div>
//                 </div>
//               </main>
//             )}
//             {category === 'people' && (
//               <main>
//                 <div className="filter-group">
//                   <div className="input-group">
//                     <label htmlFor="name">name</label>
//                     <input className="names" name="name" type="text" />
//                   </div>
//                 </div>
//                 <div className="filter-group gender">
//                   <div className="input-group">
//                     <label>gender</label>
//                     <FontAwesomeIcon icon="male" className="male-icon" />
//                   </div>
//                   <div className="input-group">
//                     <FontAwesomeIcon icon="female" className="female-icon" />
//                   </div>
//                 </div>
//               </main>
//             )}
//           </aside>
//         </div>
//       }
//     </main>
//   )
// }
