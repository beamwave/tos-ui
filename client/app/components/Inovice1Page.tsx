import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { loadModal } from '../actions/modal'
import { GRADE_ADJUSTMENTS_MODAL } from '../constants/modaltypes'

import moment from 'moment'
import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from 'reactstrap'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

import { startTestTakerRetrieval } from '../actions/testTaker'

interface IProps {
  results: any
  loadModal: any
  testTakerData: any
  startTestTakerRetrieval: (data: any) => any
}

export class Invoice1Page extends Component<IProps> {
  state = {}

  gradeAdjustmentModal = () => this.props.loadModal(GRADE_ADJUSTMENTS_MODAL)

  componentDidMount = () => {
    const { startTestTakerRetrieval } = this.props
    // const { getDataFromStore } = this.state
    const data = { aicode: '471315' }
    startTestTakerRetrieval(data)
  }

  render = () => {
    const { results, testTakerData } = this.props

    console.log('test taker data:', testTakerData)

    return (
      <div className="invoice-1-page-container">
        <header>
          <div className="progress-bar">
            <div className="titles">
              <p>School Info</p>
              <p>Borrowed/Loaned Test Books</p>
              <p>Confirmation</p>
            </div>
            <div className="stages">
              <div className="stage" />
              <div className="stage" />
              <div className="stage" />
              <div className="line" />
            </div>
          </div>
        </header>
        <main>
          <section className="body">
            <section>
              <h2>Provide information to help us calculate your invoice</h2>
              <p>
                The information you are about to provide is used to calculate
                your invoice. Invoices will be sent to schools in mid-January.
                You must provide invoice information by 1/11/2019. You cannot
                make edits after this date.
              </p>
            </section>
            <section>
              <h2>Send bill to</h2>
              <form className="user-information-form">
                <div className="form-group">
                  <label>First Name *</label>
                  <input />
                </div>
                <div className="form-group">
                  <label>Last Name *</label>
                  <input />
                </div>
                <div className="form-group">
                  <label>Job Title *</label>
                  <select>
                    <option value="select">- Select Job Title -</option>
                    <option value="c">K12: Counselor</option>
                    <option value="tc">K12: Test Coordinator</option>
                    <option value="apc">K12: AP Coordinator</option>
                    <option value="p">K12: Principal</option>
                    <option value="ap/d">K12: Assistant Principal/Dean</option>
                    <option value="dh">K12: Department Head</option>
                    <option value="sf">K12: School Faculty</option>
                    <option value="apt">K12: AP Teacher</option>
                    <option value="t">K12: Teacher (non-AP)</option>
                    <option value="cs">K12: Curriculum Specialist</option>
                    <option value="o">K12: Other - K-12</option>
                  </select>
                </div>
                <div className="form-group">
                  <label>Email address *</label>
                  <input />
                </div>
                <div className="form-group">
                  <label>Purchase Order Number</label>
                  <input />
                </div>
              </form>
            </section>
            <section>
              <h2>Students tested</h2>
              <p>
                The numbers below show the number of standard tests shipped (for
                your reference) and the number of student answer sheets received
                and will be used to calculate your bill. If this isn't what you
                are expecting, it is possible that a student may have indicated
                an incorrect grade. Grade adjustments may be made for a limited
                time to ensure that your bill is accurate. Please note that any
                adjustments made here will not be reflected in score reports.
              </p>
              <div className="grade-adjustments">
                <p
                  className="link grade-adjustments-button"
                  onClick={this.gradeAdjustmentModal}
                >
                  Grade Adjustments
                </p>
                <p className="grade-adjustments-tooltip">
                  Grade is often used to determine who to bill for student test
                  fees (for example, whether fees are billed to the school,
                  district, or state), and occasionally students may indicate an
                  incorrect grade. Grade adjustments may be made for a limited
                  time to ensure that your bill is accurate. Please note that
                  any adjustments made here will not be reflected in score
                  reports. Also note that grade adjustments will be discarded if
                  there is a subsequent update to your school's answer sheet
                  data; any previous grade adjustments will need to be
                  re-entered.
                </p>
              </div>

              <div className="students-tested-table">
                <div className="headers">
                  <p>Grade</p>
                  <p>Tests Shipped</p>
                  <p>Ans Sheets Received</p>
                  <p>Contract</p>
                </div>
                {testTakerData.map(
                  ({ grade, shipped, ansSheets, contract }) => (
                    <div
                      className="grade-row"
                      key={`${grade}-${shipped}-${ansSheets}`}
                    >
                      <p>{grade}</p>
                      <p>{shipped}</p>
                      <p>{ansSheets}</p>
                      <p>{contract}</p>
                    </div>
                  )
                )}
                <div className="total-row">
                  <p>Total</p>
                  <p>
                    {testTakerData.reduce(
                      (store, { shipped }) => store + shipped,
                      0
                    )}
                  </p>
                  <p>
                    {testTakerData.reduce(
                      (store, { ansSheets }) => store + ansSheets,
                      0
                    )}
                  </p>
                </div>
              </div>
            </section>
            <section>
              <h2>Students covered by contracts</h2>
              <p>(table varies between psat/sat/)</p>
              <div className="contract-table">
                <div className="headers">
                  <p>Covered by</p>
                  <p>SAT (witout Essay)</p>
                  <p>SAT with Essay</p>
                </div>
                <div className="state">
                  <p>State Contract</p>
                  <p>0</p>
                  <p>0</p>
                </div>
                <div className="district">
                  <p>District Contract</p>
                  <p>0</p>
                  <p>0</p>
                </div>
              </div>
            </section>
          </section>
          <aside>
            <div className="sidepanel-container">
              {/* <h3>{name}</h3>
            <p>{aicode}</p>
            <p>{admin}</p>
            <p>{date}</p> */}
              <h3>Princess Anne high school</h3>
              <p>(471315)</p>
              <p>Sat School Day</p>
              <p>Wed. October 10, 2018</p>
              <p>Reason for modifying or submitting invoice</p>
              <textarea />
              <Link to="/invoice" className="blue-link-button">
                Continue
              </Link>
              <Link to="/school-landing" className="link">
                exit
              </Link>
            </div>
          </aside>
        </main>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  results: state.school.results[0], // change to know which school was clicked on
  testTakerData: state.testTaker.data
})

export default connect(
  mapStateToProps,
  { startTestTakerRetrieval, loadModal }
)(Invoice1Page)
