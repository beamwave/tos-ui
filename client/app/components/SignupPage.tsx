import React, { Component } from 'react'
import { Link, RouteComponentProps } from 'react-router-dom'
import { FontAwesomeIcon } from '../../../node_modules/@fortawesome/react-fontawesome'

import { connect } from 'react-redux'
import { startSignup } from '../actions/auth'
import { generateUuid } from '../helpers/helpers'

interface IProps {
  history?: any
  startSignup(data: {}): any
}

interface IState {
  username: string
  email: string
  password: string
  message: string
  admin: boolean
  errors: {
    username: string
    email: string
    password: string
  }
}

export class SignupPage extends Component<IProps, IState> {
  state = {
    email: '',
    fullname: '',
    username: '',
    password: '',
    message: '',
    admin: false,
    errors: {
      username: '',
      email: '',
      password: '',
      global: ''
    }
  }

  // @ts-ignore
  onFieldChange = e => {
    const { name, value } = e.target
    this.setState({
      [name]: value
    } as any)
  }

  // @ts-ignore
  onSubmit = async e => {
    e.preventDefault()
    const { startSignup, history } = this.props
    const personName = this.state.fullname.split(' ')

    // single variable used in cognito signup
    const { email, username, password, admin } = this.state

    const data = {
      email: this.state.email,
      firstname: personName[0],
      lastname: personName[personName.length - 1],
      username: this.state.username,
      password: this.state.password
      // uid: generateUuid()
      // role: admin ? 'admin' : 'employee'
    }

    try {
      console.log('starting signup')
      const res = await startSignup(data)
      console.log('login success', res)
      history.push('/dashboard')
    } catch (err) {
      console.log('login failure', err)
      this.setState({ errors: err.response.data.errors })
    }
  }

  // @ts-ignore
  render = () => {
    const { errors, admin, message } = this.state
    return (
      <div className="signup-page">
        <div className="signup-bg" />
        <Link to="/" className="header">
          <div className="globe">
            <FontAwesomeIcon icon="globe" />
          </div>
          <h1 className="app-name">Arsenal</h1>
        </Link>
        <form
          autoComplete="off"
          className="signup-form"
          onChange={this.onFieldChange}
          onSubmit={this.onSubmit}
        >
          <div className="input-group">
            <label htmlFor="email" className="title">
              email
            </label>
            <input type="email" name="email" />
          </div>
          <div className="input-group">
            <label htmlFor="fullname" className="title">
              full name
            </label>
            <input type="text" name="fullname" />
          </div>
          <div className="input-group">
            <label htmlFor="username" className="title">
              username
            </label>
            <input type="text" name="username" />
          </div>
          <div className="input-group">
            <label htmlFor="password" className="title">
              password
            </label>
            <input
              type="password"
              name="password"
              autoComplete="new-password"
            />
          </div>
          <div className="input-group">
            <div className="switch-page">
              <p className="text">Already have an account?</p>
              <Link to="/login" className="login-link">
                &nbsp;Login
              </Link>
            </div>
            <button type="submit" className="signup-button button">
              Signup
            </button>
          </div>
        </form>
        {!!errors.global && <p>{errors.global}</p>}
        {message && <div className="message">{message}</div>}
        <div className="signup-overlay" />
        <div className="signup-bg" />
      </div>
    )
  }
}

export default connect(
  null,
  { startSignup }
)(SignupPage)
