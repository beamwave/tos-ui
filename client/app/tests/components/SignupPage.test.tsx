import React, { Component } from 'react'
import { shallow } from 'enzyme'
import { SignupPage } from '../../components/SignupPage'

test('should render SignupPage component', () => {
  const wrapper = shallow(
    <SignupPage startSignup={() => `placeholder function`} />
  )
  expect(wrapper).toMatchSnapshot()
})
