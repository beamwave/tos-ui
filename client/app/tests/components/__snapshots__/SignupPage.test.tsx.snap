// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SignupPage component 1`] = `
<div
  className="signup-page"
>
  <div
    className="signup-bg"
  />
  <Link
    className="header"
    replace={false}
    to="/"
  >
    <div
      className="globe"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="globe"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
    </div>
    <h1
      className="app-name"
    >
      Arsenal
    </h1>
  </Link>
  <form
    autoComplete="off"
    className="signup-form"
    onChange={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <label
        className="title"
        htmlFor="email"
      >
        email
      </label>
      <input
        name="email"
        type="email"
      />
    </div>
    <div
      className="input-group"
    >
      <label
        className="title"
        htmlFor="fullname"
      >
        full name
      </label>
      <input
        name="fullname"
        type="text"
      />
    </div>
    <div
      className="input-group"
    >
      <label
        className="title"
        htmlFor="username"
      >
        username
      </label>
      <input
        name="username"
        type="text"
      />
    </div>
    <div
      className="input-group"
    >
      <label
        className="title"
        htmlFor="password"
      >
        password
      </label>
      <input
        autoComplete="new-password"
        name="password"
        type="password"
      />
    </div>
    <div
      className="input-group"
    >
      <div
        className="switch-page"
      >
        <p
          className="text"
        >
          Already have an account?
        </p>
        <Link
          className="login-link"
          replace={false}
          to="/login"
        >
          Â Login
        </Link>
      </div>
      <button
        className="signup-button button"
        type="submit"
      >
        Signup
      </button>
    </div>
  </form>
  <div
    className="signup-overlay"
  />
  <div
    className="signup-bg"
  />
</div>
`;
